app/controllers/api/v1/migrations/external_controller.rb

```diff
@@ -38,17 +42,19 @@ def create_role
         # Does: Creates a user.
 
         def create_user
-          language = if user_params[:language].blank? || user_params[:language] == 'default'
-                       I18n.default_locale
-                     else
-                       user_params[:language]
-                     end
+          user_hash = user_params.to_h
+
+          return render_data status: :created if User.exists? email: user_hash[:email], provider: 'greenlight'
 
-          role = Role.find_by(name: user_params[:role], provider: 'greenlight')
+          user_hash[:language] = I18n.default_locale if user_hash[:language].blank? || user_hash[:language] == 'default'
+
+          role = Role.find_by(name: user_hash[:role], provider: 'greenlight')
 
           return render_error status: :bad_request unless role
 
-          user = User.new(user_params.merge(provider: 'greenlight', password: generate_secure_pwd, language:, role:))
+          user_hash[:password] = generate_secure_pwd if user_hash[:external_id].blank?
+
+          user = User.new(user_hash.merge(provider: 'greenlight', role:))
```

Comment:
Another approach is the following:
```
          user = User.create(user_hash.merge(provider: 'greenlight', role:))

          return render_data status: :created if user.errors.size == 1 && user.errors.of_kind?(:email, :taken)

          return render_error status: :bad_request unless user.persisted?
```
This offers the following advantages:
1. For initial requests it will have less queries (one less to be specific, around 0.5 ms).

But suffers from the following disadvantages:
1. It requires more boilerplate code and complexity.
2. For all duplicated requests it will cause way more queries and logic to fire to catch the duplicated attempt.